spec/02-integration/05-proxy/09-balancer_spec.lua

```diff
@@ -1331,37 +1331,126 @@ for _, strategy in helpers.each_strategy() do
 
           describe("with consistent hashing", function()
 
-            it("over multiple targets", function()
-              local requests = SLOTS * 2 -- go round the balancer twice
+            describe("over multiple targets", function()
+
+              it("hashing on header", function()
+                local requests = SLOTS * 2 -- go round the balancer twice
+
+                local upstream_name = add_upstream({
+                  hash_on = "header",
+                  hash_on_header = "hashme",
+                })
+                local port1 = add_target(upstream_name, localhost)
+                local port2 = add_target(upstream_name, localhost)
+                local api_host = add_api(upstream_name)
+
+                -- setup target servers
+                local server1 = http_server(localhost, port1, { requests })
+                local server2 = http_server(localhost, port2, { requests })
+
+                -- Go hit them with our test requests
+                local oks = client_requests(requests, {
+                  ["Host"] = api_host,
+                  ["hashme"] = "just a value",
+                })
+                assert.are.equal(requests, oks)
+
+                -- collect server results; hitcount
+                -- one should get all the hits, the other 0
+                local _, count1 = server1:done()
+                local _, count2 = server2:done()
+
+                -- verify
+                assert(count1 == 0 or count1 == requests, "counts should either get 0 or ALL hits")
+                assert(count2 == 0 or count2 == requests, "counts should either get 0 or ALL hits")
+                assert(count1 + count2 == requests)
+              end)
+
+              describe("hashing on cookie", function()
+                it("when the cookie is set", function()
```

Comment:
I can't think of any edge cases that testing when a cookie _is_ set catch that are not caught in the next example (where the cookie is _not_ set). I would favor removing this test as redundant, but defer to your judgement.

Reply:
agreed. But there is no test that validates that no cookie will be set if it was already provided, so maybe update this test to do that?

Reply:
note: it is good practice to write the intended result of a test in its description, rather than relying on the reader to infer the proper test results by reading the test case... In this case, a name such as `does not reply with Set-Cookie if cookie is already set` or something.